src/libcore/num/mod.rs

```diff
@@ -1619,6 +1620,50 @@ $EndFeature, "
             #[inline]
             pub fn is_negative(self) -> bool { self < 0 }
         }
+
+        /// Return the memory representation of this integer as a byte array.
+        ///
+        /// The target platform’s native endianness is used.
+        /// Portable code likely wants to use this after [`to_be`] or [`to_le`].
+        ///
+        /// [`to_be`]: #method.to_be
+        /// [`to_le`]: #method.to_le
+        ///
+        /// # Examples
+        ///
+        /// ```
+        /// #![feature(int_to_from_bytes)]
+        ///
+        /// let bytes = i32::min_value().to_be().to_bytes();
+        /// assert_eq!(bytes, [0x80, 0, 0, 0]);
+        /// ```
+        #[unstable(feature = "int_to_from_bytes", issue = "49792")]
+        #[inline]
+        pub fn to_bytes(self) -> [u8; mem::size_of::<Self>()] {
+            unsafe { mem::transmute(self) }
```

Comment:
This doesn't need to be a transmute right? Can you do something like this instead?

```rust
unsafe { *(&self as *const Self as *const u8 as *[u8; mem::size_of::<Self>()]) }
```

Maybe we don't care about decreasing the use of transmute?

Also, neat, I didn't realize `mem::size_of` could be used at the type level.

Reply:
It could be that, but that seems harder to read to me. Why would it be preferable? Regarding `transmute` I often prefer to fully qualify it with a turbofish, but in this case that felt unnecessary since it forms the entire body of a method that has an explicitly-written signature.

`size_of` became a `const fn` in https://github.com/rust-lang/rust/pull/42859

Reply:
I think transmute is fine here, because we're transmuting non-reference types. For reference/pointer types I agree that we should not be using transmute (this is what we suggest in clippy at least)

Reply:
Sounds good to me!

Reply:
@oli-obk I’m not opposing this principle, but could you say more about why?